objects/objPauseMenu.object.gmx

```diff
@@ -105,14 +105,48 @@ with objMegaman event_user(0);
                         canMove = true;
                     }
                 }
-                playSFX(sfxMenuSelect);
+                
+                // If an item hasn't been primed for use, transition to phase 3.
+                if (primedItemIndex == -1)
+                {
+                    playSFX(sfxMenuSelect);
+                    phase = 3;
+                }
+                // Otherwise, transition to phase 5.
+                else
+                {
+                    global.items[item_index].using = true;
+                    global.items[item_index].useOnIndex = option;
+                    option = 99;
+                    phase = 5;
+                }
             }
             else {  //Selected an item
                 item_index = option - global.totalWeapons;
-                with global.items[item_index] event_user(0);    //Activate item
-                if global.items[item_index].ok {
-                    option = 99;
-                    phase = 5;
+                
+                // If the item selected was already primed, unprime it.
+                if (item_index == primedItemIndex)
+                {
+                    with global.items[item_index] event_user(3);    //Unprime item.
+                    primedItemIndex = -1;
+                }
+                else
+                {
+                    with global.items[item_index] event_user(0);    //Activate item
```

Comment:
The issue I mentioned in the comments probably must be addressed here. If primed > -1, you can't activate the item.